### ISSUE TITLE
Disable ui-grid cell edit on the fly.

### ISSUE THREAD

**Issue by uguryilmaz** [angular-ui/ng-grid#3726]
hi, is there a way to disable/enable cell edit of the grid. for example i have a master-detail form screen. when user wants to analyze a record, i disable all controls of the form with ng-disabled. but i could not prevent editing the grid. i tried "cellEditableCondition" option. But, i runs only when grid is loading. it would be nice if grid have an option like "disableEdit" and accepts a scope variable. when i open my form in edit mode, grid would be editable, and when in view mode grid would be disabled.

**Comment by PaulL1** [comment]
I think you can do this on the columnDefs.

**Comment by uguryilmaz** [comment]
thank you for response. yes i tried it on columnDefs but it works only when thr form loading. after that i can not re enable (or disable) the grid. when the form first loaded the grid is in not-editable mode. (cellEditableCondition = false) there is an edit button on the form screen. when edit button clicked i want to make the grid editable (cellEditableCondition = true). i used a scope variable such as $scope.formProperties.isEditing but it didnt worked.

**Comment by PaulL1** [comment]
On the columnDef, you want to use `enableCellEdit`: http://ui-grid.info/docs/#/tutorial/201_editable

I think this is able to be dynamically changed, you may need to call notifyDataChange to make the grid notice that you've changed it.

**Comment by uguryilmaz** [comment]
i tried notifyDataChanged but not worked. 

      //on grid initialization i disable cell edit.
      $scope.EntityDetailGrid.enableCellEdit = false;

      //then when i click on update button i set enableCellEdit and call notifyDataChange
      $scope.UpdateDetail = function () {
            $scope.entityProperties.pageProperties.isView = false;
            $scope.EntityDetailGrid.enableCellEdit = $scope.entityProperties.pageProperties.isView;
            $scope.EntityDetailGrid.currentScope.gridApi.core.notifyDataChange($scope.EntityDetailGrid.currentScope.gridApi.grid, uiGridConstants.dataChange.ALL);
        };

what am i doing wrong?

by the way, i found a plunker about a sample similar to my problem. he/she used notifyDataChange function and it isnt working too. 

he/she says, found a solution for his/her problem but its sample is not working. 

this is the thread about notifyDataChanged:

https://github.com/angular-ui/ng-grid/issues/2598

and this is the plunker:

http://plnkr.co/edit/AzyY6zfB39jFdWl0XhiZ?p=preview

when you click on the item on column menu of Name column (Toggle Color) it must be clear colors and click againg, it must set colors back, but colors not clearing.

**Comment by PaulL1** [comment]
`enableCellEdit` gets set on the columnDef, not on the gridOptions.  Check the tutorial I provided a link to.

**Comment by uguryilmaz** [comment]
hi paul, thank you for your help. i did what you said but still not working. here is what i have done.

first if you look $scope.EntityDetailGrid.columnDefs[0].enableCellEdit it is false.. 

![image](https://cloud.githubusercontent.com/assets/6348819/8052756/0a834360-0e93-11e5-9fbf-3b3c40f7380c.png)

i set it to "true" and then call notify data change.

![image](https://cloud.githubusercontent.com/assets/6348819/8052797/7a00d4c8-0e93-11e5-9473-60331e99a75a.png)

i tried uiGridConstants.dataChange.EDIT, uiGridConstants.dataChange.ALL, uiGridConstants.dataChange.COLUMN but nothing happened.

**Comment by PaulL1** [comment]
Try this: http://plnkr.co/edit/xit2sdognj9Z01DpFU5N?p=preview

Looking at the code base, I don't think colDef.enableCellEdit is dynamic - it looks to be evaluated once at startup.

**Comment by uguryilmaz** [comment]
thank you so much paul. that was what i want. thank you for your help.

**Comment by tandvu** [comment]
My project uses View-Controller-Model.

In my controller:

function MyController(scope, filter, uiGridConstants) 
{
       var vm = this;
        scope.edit = false;
        scope.canEdit = function(){
            return scope.edit;
        }

        scope.gridOptions = {
            enableSorting: true,
            columnDefs: [
                { field: 'firstName', displayName: "First Name", width: 220, cellEditableCondition : scope.canEdit },
                { field: 'company', cellEditableCondition : scope.canEdit}
            ],
            onRegisterApi: function( gridApi ) {
                scope.gridApi = gridApi;
                scope.gridApi.core.on.sortChanged(scope, function( grid, sort ) {
                scope.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN );
                })
            },
            data: vm.prop.myData
        };
}

It works fine if I use scope.  However, if I change all scope to vm, the cellEditableCondition stops working.  It means I still have the table display all the data, but I can't turn on/off edit mode.

**Comment by imbalind** [comment]
@tandvu this issue is already closed, furthermore your problem is quite different from the original one. I suggest you to ask this kind of question on a q&a website like Stack Overflow. Remember to tag your question with angular-ui-grid

**Comment by sumeetsonawane11** [comment]
I have a grid which have multi-select checkbox input field [on clicking input field, i get a dropdown values and after selecting , the value is stored ], I have a edit icon on the right side.

This is **CellTemplate:**
When the page loads , all the fields are in non-editable mode [page-load_noneditable.png] and when I clicked on edit icon the fields are in edit mode [enablecelledit = true]. This is [] 

This is **enableCellTemplate**
In the edit mode when I click on the multidropdown field and select one of the values from the options , that values gets displayed in the field[editable.png]. This is different template []

After this, when I click plus (+) icon to save the edited changes ,  my CellTemplates get rendered for all the fields except the multi-dropdown field. It is still in the edit mode and shows the values.

I have attached some screenshots for reference.

![non-edit](https://user-images.githubusercontent.com/22935051/42282770-35d9b58e-7fc5-11e8-8474-545f98348dd3.PNG)
![editable](https://user-images.githubusercontent.com/22935051/42282771-36253d2e-7fc5-11e8-8208-17b3c08c54b3.PNG)
![page-load_noneditable](https://user-images.githubusercontent.com/22935051/42282772-366be210-7fc5-11e8-985d-517ecd45d2f1.PNG)
